File: superstitioMain/java/superstitio/cards/maso/SkillCard/HeavyCuffs.java
package superstitio.cards.maso.SkillCard;

import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cardModifier.modifiers.block.WhenLostDrawCardBlock;
import superstitio.cards.maso.MasoCard;


public class HeavyCuffs extends MasoCard {
    public static final String ID = DataManager.MakeTextID(HeavyCuffs.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.COMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 1;
    private static final int MAGIC = 2;
    private static final int UPGRADE_MAGIC = 1;
    private static final int BLOCK = 8;
    private static final int UPGRADE_BLOCK = 3;

    public HeavyCuffs() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
        this.setupBlock(BLOCK, UPGRADE_BLOCK, new WhenLostDrawCardBlock(this.magicNumber));
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot_gainBlock();
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/maso/SkillCard/HeavyCuffs.kt
package superstitio.cards.maso.SkillCard

import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cardModifier.modifiers.block.WhenLostDrawCardBlock
import superstitio.cards.IsMasoCard
import superstitio.cards.maso.MasoCard
@IsMasoCard
class HeavyCuffs : MasoCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
        this.setupBlock(BLOCK, UPGRADE_BLOCK, WhenLostDrawCardBlock(this.magicNumber))
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot_gainBlock()
    }

    override fun upgradeAuto()
    {
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(HeavyCuffs::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.COMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 1
        private const val MAGIC = 2
        private const val UPGRADE_MAGIC = 1
        private const val BLOCK = 8
        private const val UPGRADE_BLOCK = 3
    }
}
